src/Core/BackendAggregator.vala

```diff
@@ -0,0 +1,151 @@
+/*-
+ * Copyright 2019 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: David Hewitt <davidmhewitt@gmail.com>
+ */
+
+public class AppCenterCore.BackendAggregator : Backend, Object {
+    public Gee.ArrayList<unowned Backend> backends;
```

Comment:
While we're here, I don't see this array used anywhere in the code, at least not publicly by some other object. It is probably safe to remove this or at least have it return a read only view to the list.